schemas/derivatives/SCHEMA.md

```diff
@@ -0,0 +1,83 @@
+# Derivatives
+
+OpenBlock Labs standard schema for derivatives protocols (options and perps).
+
+## Version: 1.0.0-alpha
+
+### Pools
+
+Pools in the protocol (one entry for each token).
+
+| Property                | Description                                               | Type   |
+|-------------------------|-----------------------------------------------------------|--------|
+| chain_id                 | Standard chain id.                                        | number |
+| creation_timestamp       | The timestamp of the block the pool was created on.       | number |
+| creation_block_number    | The block number this pool was created on.                | number |
+| pool_address             | The contract address of the pool.                         | string |
+| pool_name                | The name of the pool (ie, name() in the smart contract).  | string |
+| token_index              | The index of the token in the smart contract (one row for each token in a pool). | number |
+| token_address            | The contract address of the token.                        | string |
+| token_symbol             | The symbol of the token.                                  | string |
+
+### LP Snapshot
+
+Liquidity providers snapshot (one entry for each token in a pool).
+
+| Property                | Description                                               | Type   |
+|-------------------------|-----------------------------------------------------------|--------|
+| timestamp                | The timestamp of the snapshot.                            | number |
+| chain_id                 | Standard chain id.                                        | number |
+| pool_address             | The address of the pool.                                  | string |
+| lp_address               | The address of the liquidity provider.                    | string |
+| token_index              | The index of the token in the smart contract.             | number |
+| token_address            | The address of the token provided as liquidity.           | string |
+| token_symbol             | The symbol of the token.                                  | string |
+| deposit_amount           | (Optional, for Order Books) The amount deposited into the pool, decimal normalized. | number |
+| deposit_amount_usd       | (Optional, for Order Books) The amount deposited, in USD. | number |
+| amount_useful            | (Optional, for LPs) The amount of tokens supplied that are liquid and usable by users of the protocol, decimal normalized. | number |
+| amount_useful_usd        | (Optional, for LPs) The amount of liquid tokens supplied in USD. | number |
```

Comment:
should this be calculated on user level ? seems difficult to do so.

eg : for protocols that forked GMX's model , how do we calculate the amount_useful here for GLP?

Reply:
or this is suppose to be the available margin that the user have on the wallet?

total balance - margin used in active positions?

Reply:
user level would be more useful indeed. 
For GMX-like protocols, can we take the GLP minted and substract the burnt ones when redeemed? 

balence-Margin used is at the taker level. But if there's no liquidity on the maker side, it's difficult to make such trades, so we'd need both sides. My take is that this part of the schema is for the maker side, but temporary. we should have a schema that encompasses makers both for AMMs and order book style protocols.

Reply:
> user level would be more useful indeed.
For GMX-like protocols, can we take the GLP minted and substract the burnt ones when redeemed?

example : 

if user deposits 1k into GLP/GM pool (pool has 1m now)
there is OI of 200k.

is this `amount_useful` to be :
1) 1k
2) 1k  * (200k / 1mm) = 800?

if its 1) then i think thats okay

Reply:
we can use 1k for the time being, although it's inaccurate because we don't know whether it's in range or not. That's why standalone maker schema makes more sense, that can be integrated.